src/Mod/PartDesign/Gui/TaskExtrudeParameters.h

```diff
@@ -84,80 +103,107 @@ class TaskExtrudeParameters : public TaskSketchBasedParameters
     void fillDirectionCombo();
     void addAxisToCombo(App::DocumentObject* linkObj, std::string linkSubname, QString itemText,
         bool hasSketch = true);
-    void applyParameters(QString facename);
+    void applyParameters();
 
     void setSelectionMode(SelectionMode mode);
 
 protected Q_SLOTS:
     void onLengthChanged(double);
     void onLength2Changed(double);
     void onOffsetChanged(double);
+    void onOffset2Changed(double);
     void onTaperChanged(double);
     void onTaper2Changed(double);
     void onDirectionCBChanged(int);
     void onAlongSketchNormalChanged(bool);
     void onDirectionToggled(bool);
     void onAllFacesToggled(bool);
+    void onAllFaces2Toggled(bool);
     void onXDirectionEditChanged(double);
     void onYDirectionEditChanged(double);
     void onZDirectionEditChanged(double);
-    void onMidplaneChanged(bool);
     void onReversedChanged(bool);
     void onFaceName(const QString& text);
+    void onFaceName2(const QString& text);
     void onSelectFaceToggle(const bool checked = true);
     void onSelectShapeToggle(const bool checked = true);
     void onSelectShapeFacesToggle(const bool checked);
     void onUnselectShapeFacesTrigger();
+    void onSelectFace2Toggle(const bool checked = true);
+    void onSelectShape2Toggle(const bool checked = true);
+    void onSelectShapeFaces2Toggle(const bool checked);
+    void onUnselectShapeFaces2Trigger();
 
+    void onSidesModeChanged(int);
     virtual void onModeChanged(int);
+    virtual void onMode2Changed(int);

```

Comment:
that is a lot of repetition - maybe you should extract additional widget that would contain that logic?